code/lib/test/src/index.ts

```diff
@@ -11,10 +11,19 @@ import {
   resetAllMocks,
   restoreAllMocks,
 } from './spy';
-import type { Renderer } from '@storybook/types';
+import { type queries, within, userEvent } from './testing-library';
 
 export * from './spy';
 
+type Queries = ReturnType<typeof within<typeof queries>>;
+
+declare module '@storybook/csf' {
+  interface Canvas extends Queries {}
+  interface StoryContext {
+    userEvent: ReturnType<typeof userEvent.setup>;
+  }
```

Comment:
Do we want to add `canvas` here as well?

Reply:
Canvas is a special type in CSF, that gets augmented  here:

https://github.com/storybookjs/storybook/blob/b9f4f0401bcc479b27d3250d483e54c7a9a78c31/code/lib/test/src/index.ts#L19-L26

And therefore the StoryContext changes as well:

```tsx
export interface StoryContext<TRenderer extends Renderer = Renderer, TArgs = Args>
  extends StoryContextForEnhancers<TRenderer, TArgs>,
    Required<StoryContextUpdate<TArgs>> {
  loaded: Record<string, any>;
  abortSignal: AbortSignal;
  canvasElement: TRenderer['canvasElement'];
  hooks: unknown;
  originalStoryFn: StoryFn<TRenderer>;
  viewMode: ViewMode;
  step: StepFunction<TRenderer, TArgs>;
  context: this;
  canvas: Canvas;
}
```

Reply:
I don't think `canvas` should be part of the CSF spec, to be honest. That makes CSF dependent on Testing Library which I don't think is desirable.

Reply:
Right, it is not coupled, because canvas is empty interface in CSF, that can be implented by addons such as `@storybook/test`.